class/command/CreateNomination.php

```diff
@@ -93,14 +91,75 @@ public function execute(Context $context)
          * Check  fields *
         *****************/
 
+        if (isset($_SESSION['redirect']) && $_SESSION['redirect'] == 'true')
+        {
+            \NQ::simple('nomination', NotificationView::NOMINATION_ERROR, 'Be sure to fill in all required fields.');
+            unset($_SESSION['redirect']);
+        }
+        
+
         // Check for missing required fields
         foreach($required as $key=>$value){
             if(!isset($context[$value]) || $context[$value] == ""){
                 $missing[] = $value;
+                $context['after'] = 'NominationForm';
+                $fields = array('nominee_banner_id'=>$context['nominee_banner_id'],
+                        'nominee_first_name'=>$context['nominee_first_name'],
+                        'nominee_middle_name'=>$context['nominee_middle_name'],
+                        'nominee_last_name'=>$context['nominee_last_name'],
+                        'nominee_asubox'=>$context['nominee_asubox'],
+                        'nominee_email'=>$context['nominee_email'],
+                        'nominee_position'=>$context['nominee_position'],
+                        'nominee_department_major'=>$context['nominee_department_major'],
+                        'nominee_years'=>$context['nominee_years'],
+                        'nominee_phone'=>$context['nominee_phone'],
+                        'nominee_gpa'=>$context['nominee_gpa'],
+                        'nominee_class'=>$context['nominee_class'],
+                        'nominee_responsibility'=>$context['nominee_responsibility'],
+                        'nominator_first_name'=>$context['nominator_first_name'],
+                        'nominator_middle_name'=>$context['nominator_middle_name'],
+                        'nominator_last_name'=>$context['nominator_last_name'],
+                        'nominator_address'=>$context['nominator_address'],
+                        'nominator_phone'=>$context['nominator_phone'],
+                        'nominator_email'=>$context['nominator_email'],
+                        'nominator_relationship'=>$context['nominator_relationship'],
+                        'category'=>$context['category'],
+                        'statement'=>array('name'=>$_FILES['statement']['name'],
+                                'type'=>$_FILES['statement']['type'],
+                                'tmp_name'=>$_FILES['statement']['tmp_name'],
+                                'error'=>$_FILES['statement']['error'],
+                                'size'=>$_FILES['statement']['size']),
+                        'reference_id'=>array('0'=>$context['reference_id_0'],
```

Comment:
This will probably only work for three references, and wouldn't repopulate anything after the third reference. Since the number of references is totally configurable, we should probably make this work for _n_ references, where _n_ is taken from the current config for number of references.